synthesizer/program/src/logic/instruction/operand/parse.rs

```diff
@@ -28,6 +28,13 @@ impl<N: Network> Parser for Operand<N> {
             map(tag("self.caller"), |_| Self::Caller),
             map(tag("block.height"), |_| Self::BlockHeight),
             map(tag("network.id"), |_| Self::NetworkID),
+            // Note that `Operand::Checksum` and `Operand::Edition` must be parsed before `Operand::ProgramID`s, since an edition or checksum may be prefixed with a program ID.
+            map(pair(opt(terminated(ProgramID::parse, tag("/"))), tag("checksum")), |(program_id, _)| {
+                Self::Checksum(program_id)
+            }),
+            map(pair(opt(terminated(ProgramID::parse, tag("/"))), tag("edition")), |(program_id, _)| {
+                Self::Edition(program_id)
+            }),
             // Note that `Operand::ProgramID`s must be parsed before `Operand::Literal`s, since a program ID can be implicitly parsed as a literal address.
```

Comment:
Is this comment out of date? Literal::parse() doesn't handle program IDs, which are handled by the explicit ProgramID::parse() below.

Reply:
`Literal::parse` does handle `ProgramID`s [[source]](https://github.com/ProvableHQ/snarkVM/blob/06964e8c81ee81ff8c0f2062234196151d6767ba/console/program/src/data/literal/parse.rs#L41).
This was needed to allow users to pass in program IDs as address value arguments but also allow program IDs to be explicitly declared as `Operand`s.

Reply:
Ah, I looked at [the Literal::parse() under circuits](https://github.com/ProvableHQ/snarkVM/blob/06964e8c81ee81ff8c0f2062234196151d6767ba/circuit/program/src/data/literal/mod.rs#L158), instead of the one under console. (I didn't know/remember there were two.)